common/functions/src/scalars/semi_structureds/check_json.rs

```diff
@@ -0,0 +1,139 @@
+    }
+
+    fn return_type(&self, args: &[&DataTypePtr]) -> Result<DataTypePtr> {
+        if args[0].data_type_id() == TypeID::Null {
+            return Ok(NullType::arc());
+        }
+
+        return Ok(Arc::new(NullableType::create(StringType::arc())));
+    }
+
+    fn eval(&self, columns: &ColumnsWithField, input_rows: usize) -> Result<ColumnRef> {
+        let mut column = columns[0].column();
+        let mut _all_null = false;
+        let mut source_valids: Option<&Bitmap> = None;
+        if column.is_nullable() {
+            (_all_null, source_valids) = column.validity();
+            let nullable_column: &NullableColumn = Series::check_get(column)?;
+            column = nullable_column.inner();
+        }
+
+        if column.data_type_id() == TypeID::Null {
+            return NullType::arc().create_constant_column(&DataValue::Null, input_rows);
+        }
+
+        let mut builder = NullableColumnBuilder::<Vu8>::with_capacity(input_rows);
+
+        with_match_physical_primitive_type!(column.data_type_id().to_physical_type(), |$T| {
+            for _i in 0..input_rows {
+                builder.append_null()
+            }
+        }, {
+            match column.data_type_id() {
+                TypeID::Boolean => {
+                    for _i in 0..input_rows {
+                        builder.append_null()
+                    }
+                }
+                TypeID::String => {
+                    let c: &StringColumn = Series::check_get(column)?;
+                    for (i, v) in c.iter().enumerate() {
+                        if let Some(source_valids) = source_valids {
+                            if !source_valids.get_bit(i) {
+                                builder.append_null();
+                                continue;
+                            }
+                        }
+
+                        match std::str::from_utf8(v) {
+                            Ok(v) => match serde_json::from_str::<JsonValue>(v) {
+                                Ok(_v) => builder.append_null(),
+                                Err(e) => builder.append(e.to_string().as_bytes(), true),
+                            },
+                            Err(e) => builder.append(e.to_string().as_bytes(), true),
+                        }
+                    }
+                }
+                TypeID::Variant => {
+                    let c: &ObjectColumn<JsonValue> = Series::check_get(column)?;
+                    for v in c.iter() {
+                        if let JsonValue::String(s) = v {
+                            match serde_json::from_str::<JsonValue>(s.as_str()) {
+                                Ok(_v) => builder.append_null(),
+                                Err(e) => builder.append(e.to_string().as_bytes(), true),
+                            }
+                        } else {
+                            builder.append_null()
+                        }
+                    }
+                }
```

Comment:
Date16, Date32, DateTime32, DateTime64, Interval should convert to string to check json

Reply:
will date be an invalid json string?

Reply:
the date is not a valid json string, `serde_json::from_str` will return error

Reply:
The date is originally a string, and we convert this string into a number when we store it, so we need to convert it back to a string here. As this string is unquoted, so converting to JSON will always fail.
In order to simplify the processing, I think it is ok not to convert to string, just append a fixed error message directly. something like `Date32 is not a valid JSON`

Reply:
got it.

Reply:
Should we append error messages or return Err.

Reply:
append error messages